Adding check for duplicated muted word

gabcvit opened the pull request:
Closes #7472 

1. Added a check to detect if the word already exists in the array of muted words. If yes, show an error message and do not save
2. Added a new translatable string for this specific scenario
3. Added translations in German and brazilian Portuguese

### Screen recording

https://github.com/user-attachments/assets/f751265a-61ec-4693-bb1d-5c5a3ada1e7f

gabcvit committed:
Adding check for duplicated muted words #7472

- Added a check to find duplicated muted words before adding them
- If muted word is found in `moderationPrefs`. It will show an error
- Added a new string for translating

gabcvit committed:
Adding translated values for `de` and `ptbr`

gabcvit commented:
Would be nice to get some feedback on it.

cpeterso commented:
@gabcvit, I can't comment on the code change, but as a user, your fix is a much better user experience!

One suggestion: considering the "This word has already been muted" message from red to green. Entering a duplicated word is not an error and red suggests the user did something wrong or something bad happened when, in fact, the message is telling the user the word is muted, just let they wanted. 🙂

gabcvit commented:
Thanks @cpeterso!
Will look into it and see how I can improve the message look 😄 
Do you know how people can get more attention from the maintainers of the repo? I can imagine they're very busy people, but 2 weeks without any feedback feels strange to me.

cpeterso commented:
Sorry, I don't know. I've only filed a few issues here. Do the Bluesky developers have a Discord or other forum?

gabcvit committed:
feat(#7472): Upgrading muted words warning to support positive messages

- In this component, it's possible to show a warning and set it up to be shown either with an error look (red) or with a positive look (green).
- With that feature, we're now able to tell the user about the duplicated muted word being added in a positive way. For improved UX

gabcvit commented:
@cpeterso, I checked around but didn't find anything. I guess I'll just leave the PR here until someone finally decides to give us some attention. Seems that they make fairly explicit that we should expect PRs to be ignored 😢 

Anyways. On a happier note, I extended the current "error message" component to support a more positive look, I renamed it to warning and added more configuration on whether it should be a negative or positive warning. This way people can reuse it for other features as well 🚀 

Here's a video on how it looks like at the moment:

https://github.com/user-attachments/assets/6d9ac58d-be66-492a-9ba4-941fbfb1a5e4
